morpho/processors/IO/IOROOTProcessor.py

```diff
@@ -52,12 +52,28 @@ def Reader(self):
         import uproot
         for key in self.variables:
             self.data.update({str(key): []})
-        tree = uproot.open(self.file_name)[self.tree_name]
-        for data in tree.iterate(self.variables):
-            for key, value in data.items():
-                varName = key.decode("utf-8")
+        try:
+            tree = uproot.open(self.file_name)[self.tree_name]
+            for data in tree.iterate(self.variables):
+                for key, value in data.items():
+                    varName = key.decode("utf-8")
+                    self.data.update({str(varName): self.data[str(varName)] + value.tolist()})
+        except:
+            logger.info("An uproot related error was encountered. Switching to ROOT.")
+            try:
+                import ROOT
+            except ImportError:
```

Comment:
Well, if it fails import, wouldn't it fail running?

Reply:
If it fails to import ROOT? That's true. I was trying to mirror the way this is done in the `Writer` function in the same processor. How should this be handled?

Reply:
Indeed!
Thanks for spotting this.
I guess we should add a warning and raise a Import error exception: basically replace the pass with a `raise ImportError`.
I should check though that this won't break the documentation build: when building the documentation on ReadTheDocs, ROOT isn't installed, thus leading to a ImportError that we need to pass...

Hum maybe, we should let it this way for now; sorry for the alarm...

Reply:
Sounds good; we can change this in the future if it won't break the documentation build.